src/Forms/GridField/GridFieldSudoMode.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace SilverStripe\Forms\GridField;
+
+use SilverStripe\Forms\FormField;
+use SilverStripe\Forms\SudoModePasswordField;
+use SilverStripe\View\ArrayData;
+use SilverStripe\View\SSViewer;
+
+/**
+ * GridFieldSudoMode warning that the managed data is protected by SudoMode which will
+ * need to be activated before the data can be edited.
+ *
+ * This is particularly important for GridFields that contain no data, which provide
+ * no other way to activate sudo mode.
+ */
+class GridFieldSudoMode extends AbstractGridFieldComponent implements GridField_HTMLProvider
+{
+    private string $sectionTitle;
+
+    private int $columnCount;
+
+    public function __construct(string $sectionTile, int $columnCount)
+    {
+        $this->sectionTitle = $sectionTile;
+        $this->columnCount = $columnCount;
+    }
+
+    public function getHTMLFragments($gridField)
+    {
+        $form = $gridField->getForm();
+        // Do not render this component if the entire form is protected by sudo mode.
+        // If the form is protected then there will already be a SudoModePasswordField as the first form field
+        if ($form->getFormRequiresSudoMode()) {
+            return [];
+        }
+        $field = SudoModePasswordField::create();
+        $field->setCollapsed(true);
+        $field->setSectionTitle($this->sectionTitle);
+        // Form needs to be set otherwise the field will not render correctly
+        $field->setForm($form);
```

Comment:
Seems to render fine without this - is this still relevant?

Reply:
Removed